### PR TITLE
COMMS Data Loaded Check and Persist walk function change

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith Please Review

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/other/Persist.js [hunk 1]
```diff
@@ -31,38 +31,58 @@
...
+                            }
+                        }
+                }
+            }
+        });
+    }
+
     return {
         discover: discover,
+        widgetWalker: widgetWalker, // jshint ignore:line
```
Comment (GordonSmith) [inline src/other/Persist.js]: Why is this needed?

File: src/marshaller/HipieDDL.js [hunk 2]
```diff
@@ -766,6 +766,7 @@
         this.URL = dataSource.URL;
         this.databomb = dataSource.databomb;
         this.request = {};
+        this.isLoaded = false;
```
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: I think this should mirror the _renderCount that widgets have - IOW rename to _loadedCount and increment it each time its loaded.

File: src/other/Persist.js [hunk 3]
```diff
@@ -31,38 +31,58 @@
...
+                    widgetArray.forEach(function(widget) {
+                        context.widgetWalker(widget, visitor);
                     });
                     break;
             }
         }
     }
 
+    function widgetPropertyWalker(widget, filter, visitor) {
+        this.widgetWalker(widget, function(widget) {
```
Comment (GordonSmith) [inline src/other/Persist.js]: What is "this" here?

File: src/marshaller/HipieDDL.js [hunk 4]
```diff
@@ -917,6 +919,20 @@
...
+        this.dashboardArray.forEach(function(dashboard) {
+            for (var ds in dashboard.datasources) {
+                dashboardLoadedArr[i] = dashboard.datasources[ds]._loadedCount;
+                i++;
+            }
+        });
+        return dashboardLoadedArr.every(function(element) {
+            return element > 0;
+        });
+    };
```
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: rewrite this function to avoid double loop.  IOW the fist time you see a _loadedCount === 0 break and return false.

**Comment by mlzummo** [comment]
@GordonSmith Please Re-Review

**Comment by mlzummo** [comment]
breaks HTML_Marshaller deserialization fix ... also need to remove commented code

**Comment by mlzummo** [comment]
@GordonSmith Please Re-Review

**Comment by GordonSmith** [comment]
Bad commit message...

Try and keep the summary line to 50 chars, then in the body of the commit you can go as long as you like.
